### PR TITLE
Eventhandler startup enhancement

### PR DESCRIPTION
**Pull request by MaGaMeGa**

### CODE REVIEW

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 1]
```diff
@@ -80,4 +105,153 @@ protected void customDestroy() {
...
+				
+				final SystemRequestDTO subscriber = DTOConverter.convertSystemToSystemRequestDTO(subscriptionList.get(0).getSubscriberSystem());		
+				
+				try {
+					
+					final Set<SystemResponseDTO> authorizedPublishers = getAuthorizedPublishers(subscriber);
+					eventHandlerDBService.updateSubscriberAuthorization(subscriptionList, authorizedPublishers);
+					
+				} catch (ArrowheadException ex) {
+
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java:161]: missing logging

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 2]
```diff
@@ -80,4 +105,153 @@ protected void customDestroy() {
...
+			for (final List<Subscription> subscriptionList : mapOfSubscriptions.values()) {
+				
+				final SystemRequestDTO subscriber = DTOConverter.convertSystemToSystemRequestDTO(subscriptionList.get(0).getSubscriberSystem());		
+				
+				try {
+					
+					final Set<SystemResponseDTO> authorizedPublishers = getAuthorizedPublishers(subscriber);
+					eventHandlerDBService.updateSubscriberAuthorization(subscriptionList, authorizedPublishers);
+					
+				} catch (ArrowheadException ex) {
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: missing final

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 3]
```diff
@@ -80,4 +105,153 @@ protected void customDestroy() {
...
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents getAuthSubscriptionCheckUri() {
+		logger.debug("getAuthSubscriptionCheckUri started...");
+
+		final UriComponents queryUri = getQueryUri();
+		checkServiceRegistryConnection(queryUri);
+		
+		findCoreSystemServiceUri(CoreSystemService.AUTH_CONTROL_SUBSCRIPTION_SERVICE, queryUri);
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: try at least 3 (use a constant) times

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 4]
```diff
@@ -94,7 +99,33 @@ protected void customInit(final ContextRefreshedEvent event) {
...
 		
-		updateSubscriberAuthorizations();
+		int retry = 0;
+		while (retry < max_retry ) {
+			try {
+				
+				updateSubscriberAuthorizations();
+				logger.info("SubscriberAuthorizations are up to date.");
+				break;
+			} catch (Exception ex) {
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: missing final

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 5]
```diff
@@ -94,7 +99,33 @@ protected void customInit(final ContextRefreshedEvent event) {
...
+				++retry;
+				logger.info("Unsuccessful update SubscriberAuthorizations. Trys left: " + (max_retry - retry) );
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: typo : trys => tries

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java [hunk 6]
```diff
@@ -94,7 +99,33 @@ protected void customInit(final ContextRefreshedEvent event) {
...
+				
+				if (retry == max_retry) {
+					
+					logger.info("EventHandler could not start becouse of unsuccessful Subscribers Authorization: " + ex);
```
Comment (rbocsi) [inline eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: typo: becouse => because

**Review by rbocsi — CHANGES_REQUESTED** [review]
minor changes needed: typos and missing final

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java (hunk 6 above)
Comment (MaGaMeGa) [reply eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: fixed

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java (hunk 5 above)
Comment (MaGaMeGa) [reply eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: fixed

File: eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java (hunk 4 above)
Comment (MaGaMeGa) [reply eventhandler/src/main/java/eu/arrowhead/core/eventhandler/EventHandlerApplicationInitListener.java]: fixed

**Pull request merged**